file_util/inc/leatherman/file_util/file.hpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file
+ * Declares utility functions for reading data from files.
+ */
+#pragma once
+
+#include <boost/filesystem.hpp>
+
+#include <string>
+#include <stdexcept>
+#include <functional>
+#include <wordexp.h>
+#include <fstream>
+
+namespace leatherman { namespace file_util {
+
+    /// Generic file error class.
+    class file_error : public std::runtime_error {
```

Comment:
I'd lean towards replacing this with `boost::filesystem::filesystem_error` for consistency.